src/main/kotlin/org/opensearch/replication/repository/RestoreContext.kt

```diff
@@ -38,10 +39,15 @@ class RestoreContext(
         val lock = fileLocks.computeIfAbsent(fileName) { ReentrantLock() }
 
         lock.withLock {
-            val baseInput = currentFiles.computeIfAbsent(fileName) {
-                store.directory().openInput(fileName, IOContext.DEFAULT)
-            }
-            return baseInput.clone()
+            var baseInput: IndexInput? = null
+
+            store.performOp({
```

Comment:
use the `performOp` block to count the number of currently open index files

Reply:
@ankitkala 

I’ve made the changes.
However, some tests are failing in a different area:
https://github.com/opensearch-project/cross-cluster-replication/actions/runs/14349010170/job/40224076758?pr=1521
If it’s a temporary issue, would it be possible to rerun the task?

Reply:
@ankitkala yes i fixed it now, could u check again plz??

Reply:
https://github.com/opensearch-project/cross-cluster-replication/actions/runs/14349244652/job/40224713873?pr=1521

This test appears to have failed due to a temporary issue.
Would it be possible to retry it?

Reply:
https://github.com/opensearch-project/cross-cluster-replication/actions/runs/14350662245/job/40228668432?pr=1521

This test is intermittently failing, could you take a look?
It seems that it has occasionally been failing even [before](https://github.com/opensearch-project/cross-cluster-replication/actions/runs/14350662245/job/40228668432?pr=1521).


It passes without any issues on my local machine.